libqtile/backend/x11/core.py

```diff
@@ -612,6 +605,10 @@ def handle_ButtonPress(self, event) -> None:  # noqa: N802
         state = event.state
         state |= self._numlock_mask
         state &= self._valid_mask
+
+        if not event.child:  # The client's handle_ButtonPress will focus it
+            self.focus_by_click(event)
+
```

Comment:
Is there a benefit to do so? The implementation is backend-specific, e.g. the Wayland one doesn't pass around an `event`, and the X one we only want to call `focus_by_click` when it was not on a window, so that it can focus the root window instead. That last bit is what the comment is refering to: a click on a window will result on both `Core.handle_ButtonPress` and `Window.handle_ButtonPress` being called, so we only want the latter to focus that window. Hence the `if not event.child`. I can clarify that comment if needed!

Reply:
Oh right, I didn't notice the different signature. :)